src/compiler.ts

```diff
@@ -2668,9 +2669,9 @@ export class Compiler extends DiagnosticEmitter {
 
     // Compile ifTrue assuming the condition turned out true
     var thenStmts = new Array<ExpressionRef>();
-    var thenFlow = flow.fork();
+    var thenFlow = flow.forkTrueBranch(condExpr);
     this.currentFlow = thenFlow;
-    thenFlow.inheritNonnullIfTrue(condExpr);
+    
```

Comment:
Btw we pretty often have this pattern:
```ts
var someFlow = flow.forkTrueBranch(condExpr);
this.currentFlow = someFlow;
...
someFlow.freeScopedLocals();
this.currentFlow = flow; // restore old flow
```
What if make this less error-prone? And automatically call `freeScopedLocals`. For that we should refator `currentFlow` to get / set accessor and use this impl:
```ts
set currentFlow(newFlow: Flow) {
  if (this._currentFlow) this._currentFlow.freeScopedLocals(); // It already have guard if locals is empty
  this._currentFlow = newFlow;
}
```
also modify `forkTrueBranch`:
```ts
forkTrueBranch(...): Flow {
   ...
   this._currentFlow = forkedFlow;
   return forkedFlow;
}
```
Then code below will be a little bit simpler and more safety:
```ts
var someFlow = flow.forkTrueBranch(condExpr);
// don't need set this.currentFlow = someFlow;
...
this.currentFlow = flow; // restore old flow
// don't need to call someFlow.freeScopedLocals();
```
@dcodeIO wdyt?

Reply:
Another alternative solution may look like this:
```
this.currentFlow.lock(); // save prev flow
var someFlow = flow.forkTrueBranch(condExpr);
// implicitly set currentFlow to `someFlow` in forkTrueBranch

this.currentFlow.unlock(); // call someFlow.freeScopedLocals() and resore previousle saved `flow` to `currentFlow`
```

Reply:
From the discussion with Daniel, we decided picked a second option, but to do it in a separate PR. So mark this as resolved